### DISCUSSION TITLE
ZSE43 Sensor Categorization

### DISCUSSION THREAD

**Discussion by justindthomas** [zwave-js/zwave-js#4128] [Q&A] [answered]
The "shock" sensor on the ZSE43 is categorized as "Diagnostic" in Home Assistant and does not show up in the normal places (auto-generated dashboard, etc.) as a result. There doesn't seem to be anything in the device file at https://github.com/zwave-js/node-zwave-js/blob/4da51ce091b8375b4809c4f64e2d9226505a9ff8/packages/config/config/devices/0x027a/zse43.json that would specify that categorization. Am I missing anything here that could be tweaked to fix the categorization?

Or is this something that's painted with a broad brush in HA and difficult to change? Like, "all shock sensors are for tamper monitoring and accordingly lumped in to 'Diagnostic'" or something along those lines?

![zse43](https://user-images.githubusercontent.com/354734/150847894-90d0daf5-105c-4d62-b0ac-f09ad9897efa.png)

The primary sensor on that shows as "disabled" because I have that turned off in the Zwave configuration (I don't care about the `tilt` sensor - I just want the `shock`).

**Comment by AlCalzone** [comment]
Which values is the driver exposing? We just pass through what the device reports and have no influence on what Homeassistant makes of that.

**Comment by kpine** [comment, marked as answer]
The cutoff screenshot doesn't really tell much. What is the full name of the shock sensor?

I pulled up the SmartThings [device handler](https://github.com/krlaframboise/SmartThings/blob/8d367f950392ec58c53624da3dd7b71535734bf2/devicetypes/zooz/zooz-zse43-tilt-shock-xs-sensor.src/zooz-zse43-tilt-shock-xs-sensor.groovy#L60-L61) code. It looks like they are using Home Security (0x7) - Tampering, product cover removed (0x3) notifications for the vibration sensor. In that case, HA categorizes device tamper sensors as diagnostic entities.

**Reply by justindthomas** [reply]
Yes, that's correct. The auto-generated entity name is `binary_sensor.tilt_shock_xs_sensor_home_security_tampering_product_cover_removed`.

Thanks, that's what I suspected. Would a request to rethink that be best placed on the Home Assistant core repo? Even aside from this use-case, cover removal seems like an operational concern rather than diagnostic.

**Reply by kpine** [reply]
Entity categorization is unrelated to the driver, so your only recourse would be submitting an issue to HA. You might be up against the opinions of HA core devs as what diagnostic means though. This device is a bit of a special case, since tamper has a different meaning.
